### PR TITLE
Remove xfail from torch-cudf.pandas integration test

### PR DESCRIPTION
**Pull request by Matt711**
## Description

The torch test should no longer fail after #16601.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by Matt711** [comment]
We can merge this PR after #16601

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_pytorch.py
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_pytorch.py:124]: TODO: Check that the test passes

File: .github/workflows/pr.yaml
Comment (Matt711) [inline .github/workflows/pr.yaml:308]: Will remove when CI passes

**Review by jameslamb — APPROVED** [review]
Giving you a `ci-codeowners` approval, expecting a `cudf-python-codeowners` reviewer will be more aware of the potential risk / impact here.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**
